aptos-move/framework/aptos-framework/sources/aptos_governance.move

```diff
@@ -561,11 +561,18 @@ module aptos_framework::aptos_governance {
     ///
     /// WARNING: currently only used by tests. In most cases you should use `reconfigure()` instead.
     /// TODO: migrate these tests to be aware of async reconfiguration.
-    public fun force_end_epoch(aptos_framework: &signer) {
+    public entry fun force_end_epoch(aptos_framework: &signer) {
         system_addresses::assert_aptos_framework(aptos_framework);
         reconfiguration_with_dkg::finish(aptos_framework);
     }
 
+    /// `force_end_epoch()` but run as admin.
+    public entry fun force_end_epoch_by_admin(admin: &signer) acquires GovernanceResponsbility {
```

Comment:
why do we need this? isn't it able to use a script to call force_end_epoch?

Reply:
for [this test](https://github.com/aptos-labs/aptos-core/pull/12420/files#r1518417617), i don't know a better way but adding this function.

Reply:
I don't think we should add functions for tests purpose, not possible to add a script there?

Reply:
If we provider a script, the script has to be compiled bytes code. Didn't dive too deep on how to do it in this test...